### PR TITLE
refactor(kubernetes): migrate node_group to plugin framework

### PR DESCRIPTION
**Pull request by villevsv-upcloud**

### CODE REVIEW

File: internal/service/kubernetes/node_group.go [hunk 1]
```diff
@@ -2,470 +2,674 @@ package kubernetes
...
+			"ssh_keys": schema.SetAttribute{
+				MarkdownDescription: "You can optionally select SSH keys to be added as authorized keys to the nodes in this node group. This allows you to connect to the nodes via SSH once they are running.",
+				ElementType:         types.StringType,
+				Optional:            true,
+				PlanModifiers: []planmodifier.Set{
+					setplanmodifier.RequiresReplace(),
+				},
+			},
+			"storage_encryption": schema.StringAttribute{
+				Computed: true,
```
Comment (kangasta) [inline internal/service/kubernetes/node_group.go:151]: Description seems to be missing. Could also add note on how the default value is computed (inherited from cluster?)

File: internal/service/kubernetes/node_group.go [hunk 2]
```diff
@@ -2,470 +2,674 @@ package kubernetes
...
-							ValidateDiagFunc: validation.ToDiagFunc(validation.StringLenBetween(0, 255)),
+						"storage_tier": schema.StringAttribute{
+							MarkdownDescription: fmt.Sprintf("The storage tier to use. Defaults to %s", upcloud.KubernetesStorageTierMaxIOPS),
+							Optional:            true,
+							Computed:            true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+							Validators: []validator.String{
+								stringvalidator.OneOf(
```
Comment (kangasta) [inline internal/service/kubernetes/node_group.go:224]: Does this support standard yet, or is it waiting for update on the service side like the developer plans?

**Review by kangasta — APPROVED** [review]

**Comment by kangasta** [comment]
Could add CHANGELOG entry about the `custom_plan` fixes as well

File: internal/planmodifier/customplan.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+func (d *customPlanPlanModifier) Description(_ context.Context) string {
+	return "Ensures that custom_plan block is set if plan field's value is custom and vice versa."
+}
+
+func (d *customPlanPlanModifier) MarkdownDescription(ctx context.Context) string {
+	return d.Description(ctx)
+}
+
+func (d *customPlanPlanModifier) PlanModifyList(ctx context.Context, req planmodifier.ListRequest, resp *planmodifier.ListResponse) {
```
Comment (kangasta) [inline internal/planmodifier/customplan.go:26]: Do we have tests for this? Could also add those later if needed.

**Comment by villevsv-upcloud** [comment]
> Could add CHANGELOG entry about the `custom_plan` fixes as well

There were no actual changes for the custom plan logic in the end, just ported from CustomizeDiff to a plan modifier.

File: internal/service/kubernetes/node_group.go (hunk 1 above)
Comment (villevsv-upcloud) [reply internal/service/kubernetes/node_group.go]: Added with the note

File: internal/service/kubernetes/node_group.go (hunk 2 above)
Comment (villevsv-upcloud) [reply internal/service/kubernetes/node_group.go:225]: Added alongside upcloud-go-api v8.7.1

File: internal/planmodifier/customplan.go (hunk 3 above)
Comment (villevsv-upcloud) [reply internal/planmodifier/customplan.go:27]: Added a unit test for now

**Pull request merged**